Server/Utilities/SpringBoardAlerts.m

```diff
@@ -48,25 +50,77 @@ - (instancetype)init {
+                                  bundle:bundle];
+            
+            NSArray * alerts = [NSJSONSerialization
+                                JSONObjectWithData:[asset data]
+                                options:kNilOptions
+                                error:nil];
+            for (int i=0; i < alerts.count; i++) {
+                NSDictionary* alertDict = alerts[i];
+                NSString * title = [alertDict objectForKey:@"title"];
+                NSArray * buttons = [alertDict objectForKey:@"buttons"];
+                id shouldAccept = [alertDict objectForKey: @"shouldAccept"];
+                if (title == nil) {
+                    @throw [NSException
+                            exceptionWithName:@"Bad springboard-alerts JSON"
+                            reason: @"No title"
+                            userInfo:[[NSDictionary alloc]
+                                      initWithObjectsAndKeys:
+                                      @"language",
+                                      language,
+                                      @"alert",
+                                      i,
+                                      nil]];
+                }
+                if (buttons == nil) {
+                    @throw [NSException
+                            exceptionWithName:@"Bad springboard-alerts JSON"
+                            reason: @"No buttons"
+                            userInfo:[[NSDictionary alloc]
+                                      initWithObjectsAndKeys:
+                                      @"language",
+                                      language,
+                                      @"alert",
+                                      i,
+                                      nil]];
+                }
+                if (buttons.count == 0) {
+                    @throw [NSException
+                            exceptionWithName:@"Bad springboard-alerts JSON"
+                            reason: @"Zero size buttons array"
+                            userInfo:[[NSDictionary alloc]
+                                      initWithObjectsAndKeys:
+                                      @"language",
+                                      language,
+                                      @"alert",
+                                      i,
+                                      nil]];
+                }
+                if (shouldAccept == nil) {
+                    @throw [NSException
+                            exceptionWithName:@"Bad springboard-alerts JSON"
+                            reason: @"No shouldAccept"
+                            userInfo:[[NSDictionary alloc]
+                                      initWithObjectsAndKeys:@"language",
+                                      language,
+                                      @"alert",
+                                      i,
+                                      nil]];
+                }
```

Comment:
I think we should extract a validation method that raises an exception. 

That would make this code easier to read and maintain.

```
// Use a class method and add some unit tests.
+ (void)raiseIfInvalidAlert:(NSDictionary *alert) {

}
```